src/backend/sev/snp/launch/mod.rs

```diff
@@ -90,7 +95,15 @@ impl<U: AsRawFd, V: AsRawFd> Launcher<Started, U, V> {
         let launch_update_data = LaunchUpdate::from(update);
         let mut cmd = Command::from(&mut self.sev, &launch_update_data);
 
-        KvmEncRegion::new(update.uaddr).register(&mut self.vm_fd)?;
+        let kvm = Kvm::new()?;
+        let vm_fd = unsafe { libc::dup(self.vm_fd.as_raw_fd()) };
+        let vm = unsafe { kvm.create_vmfd_from_rawfd(vm_fd)? };
+
+        let memory_region = kvm_enc_region {
+            addr: update.uaddr.as_ptr() as _,
+            size: update.uaddr.len() as _,
+        };
+        vm.register_enc_memory_region(&memory_region)?;
```

Comment:
The conversation is now hidden, but still not addressed https://github.com/enarx/enarx/pull/1247#discussion_r787444573

Reply:
It was done for very good reasons: you cannot do it any other way around with just a trait abstracting away all the KVM functionality. I fixed this by replacing U with a concrete type.